smithy-model/src/main/java/software/amazon/smithy/model/transform/ModelTransformer.java

```diff
@@ -481,6 +481,62 @@ public Model changeShapeType(Model model, Map<ShapeId, ShapeType> shapeToType) {
         return new ChangeShapeType(shapeToType).transform(this, model);
     }
 
+    /**
+     * Changes the type of each given shape.
+     *
+     * <p>The following transformations are permitted:
+     *
+     * <ul>
+     *     <li>Any simple type to any simple type</li>
+     *     <li>List to set</li>
+     *     <li>Set to list</li>
+     *     <li>Structure to union</li>
+     *     <li>Union to structure</li>
+     * </ul>
+     *
+     * @param model Model to transform.
+     * @param shapeToType Map of shape IDs to the new type to use for the shape.
+     * @param synthesizeEnumNames Whether enums without names should have names synthesized if possible.
+     * @return Returns the transformed model.
+     * @throws ModelTransformException if an incompatible type transform is attempted.
+     */
+    public Model changeShapeType(Model model, Map<ShapeId, ShapeType> shapeToType, boolean synthesizeEnumNames) {
```

Comment:
Rather than pass in a boolean, for future evolution of this method, it would be nice to pass in a variadic list of ChangeShapeTypeOption values, one of which is `SYNTHESIZE_ENUM_NAMES`.